Source/WebKit/WebProcess/Plugins/PDF/PDFIncrementalLoader.mm

```diff
@@ -649,19 +650,20 @@ static void dataProviderReleaseInfoCallback(void* info)
         return 0;
     }
 
-    WTF::Semaphore dataSemaphore { 0 };
     size_t bytesProvided = 0;
-
-    RunLoop::main().dispatch([protectedLoader = Ref { *this }, position, count, buffer, &dataSemaphore, &bytesProvided] {
-        protectedLoader->getResourceBytesAtPosition(count, position, [count, buffer, &dataSemaphore, &bytesProvided](const uint8_t* bytes, size_t bytesCount) {
+    // Do not dispatch main runloop (again) and wait anymore if plugin has been destroyed.
+    if (plugin->hasBeenDestroyed())
```

Comment:
Changed condition and verified on minibrowser that it does not produce the reported regression. (screen recording attached to 108489643)